Add code quality to the package

FusRoman opened the pull request:
link to  #51 
* Add pre-commit hooks running black, isort and flake8 before commit
* Add black and flake8 workflows
* make the current code base pass the pre-commit hooks 
  * ran black, isort and flake8 to make the code flake8 compliant
* Removed a lot of unused parts of the code and unused variables

FusRoman commented:
@mseglar @fabianschussler @hashkar Hi all, I added two new workflows in the package to improve the code quality: black and flake8. Before I merge the PR, could you please approve the running of the workflows and check that everything is ok?

As I plan to make several contributions to the package, can I be added to the set of authorized developers to trigger the workflows automatically? If not, I will need to ask you every time.

mseglar commented:
Just to remember: Some variables are being deleted here because some parts of the visibility script were commented out. It is not a big issue created as these are variables like time or names. 

Besides that, I went through the code changes and I am OK with them. These can be summarized as: commented out prints removed, variables not used removed, reformatting (which is the goal of the pre-commit and the workflows).

mseglar commented:
@FusRoman seems like the tests failed :(

FusRoman commented:
Yes, I'm looking at it today and will try to fix that quickly :)

FusRoman commented:
OK, it seems ready to merge. The problem was the "astro degree globe" projection not found. I removed the ligo.skymap.plot import because flake8 marked it as unused (which is True), but the import makes the projection available globally. So, I added a noqa F401, which tells flake8 not to complain about this unused import.

mseglar commented:
probably a better fix would be to import it elsewhere where it is used?

FusRoman commented:
@mseglar I agree with you. In my last commit, I moved the import right where the projections are used in the files. In addition to the visibility in the code, it allows to mitigate the import cost. I noticed that the import takes almost one second of computing time, which is quite huge. With the import in the function, it will reduce the import cost of this module.

mseglar commented:
Thanks a lot @FusRoman for noticing and improving this issue in the code. So for the leasson learned: the import was done general which would take more time than importing it in the function itself (PointingPlotting?). We had problems with ligo.skymap in the past (incompatible packages version would break the code), so we are always very attentive to this package

FusRoman commented:
> Thanks a lot @FusRoman for noticing and improving this issue in the code. So for the leasson learned: the import was done general which would take more time than importing it in the function itself (PointingPlotting?). We had problems with ligo.skymap in the past (incompatible packages version would break the code), so we are always very attentive to this package

No, not exactly; the import will take the same compute time whether in the module top level or in the function, but adding the import in the function makes it executed only when the function is called and not when the PointingPlotting file is imported. However, it is bad practice in Python to not have all the imports of a module at the top of a file. Putting the import in the function add ~1sec of ligo.skymap.plot import every time the function is called while having the import at the top level make the 1-second call only once when the module is imported for the first time. In the case of ligo.skymap, it is special, so why not... It is okay for a plotting function to have a 1 second overhead. We don't do the plot so many times.

mseglar commented:
Ahhh sure! That makes total sense. Thanks for the explanation! Better that this specific import is just done in the function and not in all the functions of the script 👍🏼
